### ISSUE TITLE
Duplicate foreign key name can lead to duplicate inferred relationship 

### ISSUE THREAD

**Issue by raphaelvigee** [stephenafamo/bob#48]
My schema is as follow (using pgsql):

```
TABLEB -> fk_tablea -> TABLEA
TABLEC -> fk_tablea -> TABLEA
```

And it generates the following structs, the first `TABLE*` should be `TABLEC`, but is `TABLEB`:

```
type TABLEARelationshipJoins[Q dialect.Joinable] struct {
	Flow            bob.Mod[Q]
	TABLEB bob.Mod[Q]
	TABLEB bob.Mod[Q]
}
```

You can see that the first entry is partially correct (`.(TABLECSlice)`)

```
func (o *TABLEA) Preload(name string, retrieved any) error {
	if o == nil {
		return nil
	}

	switch name {
	case "TABLEB":
		rels, ok := retrieved.(TABLECSlice)
		if !ok {
			return fmt.Errorf("TABLEA cannot load %T as %q", retrieved, name)
		}

		o.R.TABLEB = rels

		return nil
	case "TABLEB":
		rels, ok := retrieved.(TABLEBSlice)
		if !ok {
			return fmt.Errorf("TABLEA cannot load %T as %q", retrieved, name)
		}

		o.R.TABLEB = rels

		return nil
	default:
		return fmt.Errorf("TABLEA has no relationship %q", name)
	}
}
```

**Comment by raphaelvigee** [comment]
The issue is because both tables have the same fk defined

**Comment by stephenafamo** [comment]
I am conflicted.

In many cases, foreign key names are unique, and so it seemed like a good idea to use it as the name of the relationship. However, it can lead to issues like this.

The fix of course is to prefix the relationship name with the schema and table name, to make it something like `schema.table.fk_something`, however, this makes it more complex to figure out what the relationship name is in case it needs to be aliased or overwritten in configuration.

**Comment by raphaelvigee** [comment]
Thats the fix i m using rn, but since some dialect support same-named fk i feel like this should be supported here too, rn it breaks pretty badly, this took me a while to figure out ^^

**Comment by stephenafamo** [comment]
I decided to prefix the relationship name with the table (or table and schema) name.

To make it easier to figure out the relationship name for configuration, I added comments in the `modelR` struct with the name of the relationship.

See a3bbc93

**Comment by stephenafamo** [comment]
Released in `v0.21.0`
